android/src/main/java/com/adyenreactnativesdk/ActionHandler.kt

```diff
@@ -53,13 +53,7 @@ class ActionHandler(
     private val configuration: ActionHandlerConfiguration
 ) : Observer<ActionComponentData> {
 
-    companion object {
-        val TAG = LogUtil.getTag()
-        const val UNKNOWN_ACTION = "UNKNOWN ACTION"
-        const val ACTION_FRAGMENT_TAG = "ACTION_DIALOG_FRAGMENT"
-        private const val BUNDLE_ACTION = "bundle_action"
-    }
-
+    private var dialog: DialogFragment? = null
```

Comment:
It's better to not hold a reference to any kind of `Fragment`s and `Activity`s. If the referenced `Fragment` gets destroyed (because it's no longer in view), but this class still holds a reference to it, then the garbage collector is not able to collect it and thus there is a memory leak.

I believe you only have this reference to dismiss the dialog later. Dismissing a dialog can best be done by the parent `Fragment` or `Activity`

Reply:
Fixed :)